product-plane-services/notification-server/src/main/java/org/opendatamesh/platform/pp/notification/server/services/DispatchService.java

```diff
@@ -0,0 +1,69 @@
+package org.opendatamesh.platform.pp.notification.server.services;
+
+import org.opendatamesh.platform.core.commons.servers.exceptions.BadRequestException;
+import org.opendatamesh.platform.pp.notification.api.resources.EventNotificationResource;
+import org.opendatamesh.platform.pp.notification.api.resources.EventResource;
+import org.opendatamesh.platform.pp.notification.api.resources.ObserverResource;
+import org.opendatamesh.platform.pp.notification.api.resources.enums.EventNotificationStatus;
+import org.opendatamesh.platform.pp.notification.api.resources.exceptions.NotificationApiStandardErrors;
+import org.opendatamesh.platform.pp.notification.server.services.proxies.NotificationObserverServiceProxy;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Slice;
+import org.springframework.stereotype.Service;
+
+@Service
+public class DispatchService {
+
+    @Autowired
+    ObserverService observerService;
+
+    @Autowired
+    EventNotificationService eventNotificationService;
+
+    @Autowired
+    EventService eventService;
+
+    @Autowired
+    NotificationObserverServiceProxy notificationObserverServiceProxy;
+
+    public void notifyAll(EventResource eventToDispatch) {
+        if(eventToDispatch == null) {
+            throw new BadRequestException(
+                    NotificationApiStandardErrors.SC400_02_EVENT_IS_EMPTY,
+                    "Event object cannot be null"
+            );
+        }
+        // Create Event
+        eventToDispatch = eventService.createResource(eventToDispatch);
+        // Find all observers (page by page) and forward the event
+        Slice<ObserverResource> observerSlice = observerService.findAllResources(PageRequest.of(0, 20));
+        notifyAllSlice(observerSlice, eventToDispatch);
+        while(observerSlice.hasNext()) {
+            observerSlice = observerService.findAllResources(observerSlice.nextPageable());
+            notifyAllSlice(observerSlice, eventToDispatch);
```

Comment:
Ok la gestione, raggrupperei tutto in una singola funzione così da rendere più chiara la letture